### PR TITLE
Drop spark-3.1.x support for spark-rapids

### PR DESCRIPTION
**Pull request by NvTimLiu**
CI part for the issue: https://github.com/NVIDIA/spark-rapids/issues/10955

We'll drop spark-3.1.x support from branch-24.08, change the default spark version string to 3.2.0

This change needs to work together with the source code update for dropping spark-3.1.x support

Will keep monitoring CI jobs status after all the related changes get merged

### CODE REVIEW

**Comment by NvTimLiu** [comment]
Draft the PR, because this change needs to work together with the source code update for dropping spark-3.1.x support

**Comment by NvTimLiu** [comment]
build

File: jenkins/version-def.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ CUDA_CLASSIFIER=${CUDA_CLASSIFIER:-"cuda11"}
 CLASSIFIER=${CLASSIFIER:-"$CUDA_CLASSIFIER"} # default as CUDA_CLASSIFIER for compatibility
 PROJECT_VER=${PROJECT_VER:-"24.08.0-SNAPSHOT"}
 PROJECT_TEST_VER=${PROJECT_TEST_VER:-"24.08.0-SNAPSHOT"}
-SPARK_VER=${SPARK_VER:-"3.1.1"}
+SPARK_VER=${SPARK_VER:-"3.2.0"}
 SPARK_VER_213=${SPARK_VER_213:-"3.3.0"}
```
Comment (pxLi) [inline jenkins/version-def.sh:36]: BTW do you know if anyone relies on the `SPARK_VER_213` somewhere? it seems an unused one

**Comment by pxLi** [comment]
can you also help remove 31x https://github.com/NVIDIA/spark-rapids/blob/branch-24.08/pom.xml#L866 from the test list? thanks

**Comment by NvTimLiu** [comment]
Drop this, replaced by https://github.com/NVIDIA/spark-rapids/pull/11159

File: jenkins/version-def.sh (hunk 1 above)
Comment (NvTimLiu) [reply jenkins/version-def.sh:36]: Yes seems it's unused

**Pull request closed without merging**